Index: feeds/packages/utils/alsa-utils/Makefile
===================================================================
--- feeds/packages/utils/alsa-utils/Makefile	(revision 28602)
+++ feeds/packages/utils/alsa-utils/Makefile	(working copy)
@@ -22,7 +22,7 @@
 define Package/alsa-utils
   SECTION:=utils
   CATEGORY:=Utilities
-  DEPENDS:=+alsa-lib +libncurses +libpthread
+  DEPENDS:=+alsa-lib +libpthread
   TITLE:=ALSA (Advanced Linux Sound Architecture) utilities
   URL:=http://www.alsa-project.org/
 endef
@@ -39,6 +39,14 @@
 		--disable-rpath \
 		--disable-alsatest \
 		--disable-xmlto \
+		--disable-alsamixer \
+		--disable-aplay \
+		--disable-arecord \
+		--disable-alsactl \
+		--disable-alsaconf \
+		--disable-alsaloop \
+		--disable-largefile \
+		--disable-nls \
 		, \
 		ac_cv_prog_ncurses5_config=no \
 		ac_cv_prog_ncursesw5_config=no \
@@ -48,15 +56,15 @@
 define Package/alsa-utils/install
 	$(INSTALL_DIR) $(1)/usr/{s,}bin
 	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/amixer $(1)/usr/bin/
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/alsamixer $(1)/usr/bin/
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/aplay $(1)/usr/bin/
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/arecord $(1)/usr/bin/
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/alsactl $(1)/usr/sbin/
+	#$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/alsamixer $(1)/usr/bin/
+	#$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/aplay $(1)/usr/bin/
+	#$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/arecord $(1)/usr/bin/
+	#$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/sbin/alsactl $(1)/usr/sbin/
 
-	$(INSTALL_DIR) $(1)/usr/share/alsa/init
-	$(INSTALL_DATA) \
-	    $(PKG_INSTALL_DIR)/usr/share/alsa/init/* \
-	    $(1)/usr/share/alsa/init/
+	#$(INSTALL_DIR) $(1)/usr/share/alsa/init
+	#$(INSTALL_DATA) \
+	#    $(PKG_INSTALL_DIR)/usr/share/alsa/init/* \
+	#    $(1)/usr/share/alsa/init/
 endef
 
 define Package/alsa-utils-tests/install
