Index: package/base-files-network/files/lib/network/config.sh
===================================================================
--- package/base-files-network/files/lib/network/config.sh	(revision 28854)
+++ package/base-files-network/files/lib/network/config.sh	(working copy)
@@ -102,7 +102,6 @@
 		touch /tmp/resolv.conf.auto
 		chmod 0644 /tmp/resolv.conf.auto
 	}
-
 	local dns
 	local add
 	for dns in "$@"; do
@@ -392,7 +391,35 @@
 			# don't stay running in background if dhcp is not the main proto on the interface (e.g. when using pptp)
 			[ "$proto1" != "$proto" ] && append dhcpopts "-n -q" || append dhcpopts "-O rootpath -R &"
 			[ "$broadcast" = 1 ] && broadcast="-O broadcast" || broadcast=
-
+			
+			case "$config" in
+				wan) wifi_hostname=$(uci get -p/var/state network.wifi.lease_hostname)
+				    if [ "$wifi_hostname" != "" ]; then
+				       kill $(cat $pidfile 2>/dev/null) 2>/dev/null
+				       udhcpc -t 5 -i "$iface" ${ipaddr:+-r $ipaddr} ${clientid:+-c $clientid} ${vendorid:+-V $vendorid} -f -q -n -p "$pidfile" $broadcast > /dev/null 2>&1
+				       kill $(cat $pidfile 2>/dev/null) 2>/dev/null
+				       wifi_lease_server=$(uci get -p/var/state network.wifi.lease_server)
+				       wan_lease_server=$(uci get -p/var/state network.wan.lease_server)
+				       if [ "$wifi_lease_server" == "$wan_lease_server" ]; then
+				         hostname=$(uci get network.$config.hostname)"-alt"
+				       fi
+				    fi
+				    ;;
+				wifi) wan_hostname=$(uci get -p/var/state network.wan.lease_hostname)
+				    if [ "$wan_hostname" != "" ]; then
+				      kill $(cat $pidfile 2>/dev/null) 2>/dev/null
+				      udhcpc -t 5 -i "$iface" ${ipaddr:+-r $ipaddr} ${clientid:+-c $clientid} ${vendorid:+-V $vendorid} -f -q -n -p "$pidfile" $broadcast > /dev/null 2>&1
+				      kill $(cat $pidfile 2>/dev/null) 2>/dev/null
+				      wifi_lease_server=$(uci get -p/var/state network.wifi.lease_server)
+				      wan_lease_server=$(uci get -p/var/state network.wan.lease_server)
+				      if [ "$wifi_lease_server" == "$wan_lease_server" ]; then
+				        hostname=$(uci get network.$config.hostname)"-alt"
+				      fi
+				    fi
+				    ;;
+			esac
+			
+			
 			$DEBUG eval udhcpc -t 0 -i "$iface" \
 				${ipaddr:+-r $ipaddr} \
 				${hostname:+-H $hostname} \
@@ -400,6 +427,7 @@
 				${vendorid:+-V $vendorid} \
 				-b -p "$pidfile" $broadcast \
 				${dhcpopts}
+			
 		;;
 		none)
 			setup_interface_none "$iface" "$config"
Index: package/base-files-network/files/usr/share/udhcpc/default.script
===================================================================
--- package/base-files-network/files/usr/share/udhcpc/default.script	(revision 28854)
+++ package/base-files-network/files/usr/share/udhcpc/default.script	(working copy)
@@ -32,6 +32,7 @@
 	local user_dns
 	local user_router
 	local user_metric
+	local multivan_enable
 
 	[ -n "$ifc" ] && {
 		old_ip="$(uci_get_state network "$ifc" ipaddr)"
@@ -64,15 +65,23 @@
 		echo "udhcpc: setting default routers: $router"
 
 		local valid_gw=""
-		for i in $router ; do
-			route add default gw $i ${user_metric:+metric $user_metric} dev $interface
-			valid_gw="${valid_gw:+$valid_gw|}$i"
-		done
+		multiwan_enable=$(uci get powertools.@system[0].multiwan)
+		if [ "$multiwan_enable" == "1" ]; then
+			for i in $router ; do
+				set_multiwan $ifc $ip $interface $i ${user_metric:+metric $user_metric}
+			done
+		else
+			for i in $router ; do
+				route add default gw $i ${user_metric:+metric $user_metric} dev $interface
+				valid_gw="${valid_gw:+$valid_gw|}$i"
+			done
 		
-		eval $(route -n | awk '
-			/^0.0.0.0\W{9}('$valid_gw')\W/ {next}
-			/^0.0.0.0/ {print "route del -net "$1" gw "$2";"}
-		')
+		
+			eval $(route -n | awk '
+				/^0.0.0.0\W{9}('$valid_gw')\W/ {next}
+				/^0.0.0.0/ {print "route del -net "$1" gw "$2";"}
+			')
+		fi
 
 		change_state network "$ifc" gateway "$router"
 	}
Index: package/base-files-network/files/etc/hotplug.d/net/10-net
===================================================================
--- package/base-files-network/files/etc/hotplug.d/net/10-net	(revision 28854)
+++ package/base-files-network/files/etc/hotplug.d/net/10-net	(working copy)
@@ -16,7 +16,7 @@
 
 	scan_interfaces
 	local cfg="$(find_config "$INTERFACE")"
-
+	
 	case "$INTERFACE" in
 		# Skip wireless parent interfaces
 		ath[0-9]|wlan[0-9]) ;;
@@ -24,7 +24,16 @@
 			# check the autoload setting
 			config_get auto "$cfg" auto
 			case "$auto" in
-				1|on|enabled) setup_interface "$INTERFACE";;
+				1|on|enabled) setup_interface "$INTERFACE"
+						case "$INTERFACE" in
+						usb[0-9])
+							
+                                              		if [ -f /etc/init.d/multiwan ]; then
+                                                		/etc/init.d/multiwan restart
+                                              		fi
+							;;
+						esac
+                                              ;;
 			esac
 		;;
 	esac
@@ -68,5 +77,8 @@
 	;;
 	remove|unregister)
 		delif
+		RESOLV_CONF="/tmp/resolv.conf.auto"
+                scan_dns "$RESOLV_CONF"
+                scan_route
 	;;
 esac
Index: package/base-files-network/files/etc/init.d/network
===================================================================
--- package/base-files-network/files/etc/init.d/network	(revision 28854)
+++ package/base-files-network/files/etc/init.d/network	(working copy)
@@ -11,6 +11,7 @@
 	setup_switch
 
 	grep -qs config /etc/config/wireless && {
+		sleep 2
 		/sbin/wifi up
 	}
 
