Index: package/dropbear/Makefile
===================================================================
--- package/dropbear/Makefile	(revision 28854)
+++ package/dropbear/Makefile	(working copy)
@@ -103,6 +103,9 @@
 	$(INSTALL_DIR) $(1)/etc/dropbear
 	touch $(1)/etc/dropbear/dropbear_rsa_host_key
 	touch $(1)/etc/dropbear/dropbear_dss_host_key
+	test ! -f files/authorized_keys || $(INSTALL_DATA) ./files/authorized_keys $(1)/etc/dropbear/authorized_keys
+	test ! -f files/dropbear_rsa_host_key || $(INSTALL_DATA) ./files/dropbear_rsa_host_key $(1)/etc/dropbear/dropbear_rsa_host_key
+	test ! -f files/dropbear_dss_host_key || $(INSTALL_DATA) ./files/dropbear_dss_host_key $(1)/etc/dropbear/dropbear_dss_host_key
 endef
 
 define Package/dropbearconvert/install
