Index: package/hotplug2/files/hotplug2.rules
===================================================================
--- package/hotplug2/files/hotplug2.rules	(revision 28854)
+++ package/hotplug2/files/hotplug2.rules	(working copy)
@@ -1,6 +1,6 @@
 $include /etc/hotplug2-common.rules
 
-SUBSYSTEM ~~ (^net$|^input$|^button$|^usb$|^ieee1394$|^block$|^atm$|^zaptel$|^tty$) {
+SUBSYSTEM ~~ (^net$|^input$|^button$|^usb$|^ieee1394$|^block$|^atm$|^zaptel$|^tty$|^sound$|^input$) {
 	exec /sbin/hotplug-call %SUBSYSTEM%
 }
 
