Index: package/iptables/Makefile
===================================================================
--- package/iptables/Makefile	(revision 28854)
+++ package/iptables/Makefile	(working copy)
@@ -430,11 +430,9 @@
 	$(LN) iptables $(1)/usr/sbin/iptables-restore
 	$(INSTALL_DIR) $(1)/usr/lib/iptables
 	(cd $(PKG_INSTALL_DIR)/usr/lib/iptables ; \
-		for m in $(patsubst xt_%,ipt_%,$(IPT_BUILTIN)) $(patsubst ipt_%,xt_%,$(IPT_BUILTIN)); do \
-			if [ -f $(PKG_INSTALL_DIR)/usr/lib/iptables/lib$$$${m}.so ]; then \
-				$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/lib$$$${m}.so $(1)/usr/lib/iptables/ ;\
-			fi; \
-		done \
+		$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/libxt_standard.so $(1)/usr/lib/iptables/; \
+		$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/libxt_state.so $(1)/usr/lib/iptables/; \
+		$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/libipt_MASQUERADE.so $(1)/usr/lib/iptables/; \
 	)
 endef
 
@@ -477,11 +475,9 @@
 define BuildPlugin
   define Package/$(1)/install
 	$(INSTALL_DIR) $$(1)/usr/lib/iptables
-	for m in $(patsubst xt_%,ipt_%,$(2)) $(patsubst ipt_%,xt_%,$(2)); do \
-		if [ -f $(PKG_INSTALL_DIR)/usr/lib/iptables/lib$$$$$$$${m}.so ]; then \
-			$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/lib$$$$$$$${m}.so $$(1)/usr/lib/iptables/ ; \
-		fi; \
-	done
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/libxt_standard.so $$(1)/usr/lib/iptables/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/libxt_state.so $$(1)/usr/lib/iptables/
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/iptables/libipt_MASQUERADE.so $$(1)/usr/lib/iptables/
 	$(3)
   endef
 
