Index: feeds/packages/utils/input-utils/Makefile
===================================================================
--- feeds/packages/utils/input-utils/Makefile	(revision 28602)
+++ feeds/packages/utils/input-utils/Makefile	(working copy)
@@ -38,7 +38,7 @@
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/input-events $(1)/usr/bin/
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/input-kbd $(1)/usr/bin/
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/lsinput $(1)/usr/bin/
+	#$(INSTALL_BIN) $(PKG_BUILD_DIR)/lsinput $(1)/usr/bin/
 
 endef
 
