Index: package/kernel/modules/sound.mk
===================================================================
--- package/kernel/modules/sound.mk	(revision 28854)
+++ package/kernel/modules/sound.mk	(working copy)
@@ -31,9 +31,7 @@
 	snd-seq-device \
 	snd-rawmidi \
 	snd-timer \
-	snd-pcm \
-	snd-mixer-oss \
-	snd-pcm-oss
+	snd-pcm 
 
 SOUNDCORE_FILES ?= \
 	$(LINUX_DIR)/sound/soundcore.ko \
@@ -43,9 +41,7 @@
 	$(LINUX_DIR)/sound/core/seq/snd-seq-device.ko \
 	$(LINUX_DIR)/sound/core/snd-rawmidi.ko \
 	$(LINUX_DIR)/sound/core/snd-timer.ko \
-	$(LINUX_DIR)/sound/core/snd-pcm.ko \
-	$(LINUX_DIR)/sound/core/oss/snd-mixer-oss.ko \
-	$(LINUX_DIR)/sound/core/oss/snd-pcm-oss.ko
+	$(LINUX_DIR)/sound/core/snd-pcm.ko
 
 define KernelPackage/sound-core
   SUBMENU:=$(SOUND_MENU)
@@ -62,10 +58,7 @@
 	CONFIG_SND_VIRMIDI \
 	CONFIG_SND_SEQ_DUMMY \
 	CONFIG_SND_SEQUENCER_OSS=y \
-	CONFIG_HOSTAUDIO \
-	CONFIG_SND_PCM_OSS \
-	CONFIG_SND_MIXER_OSS \
-	CONFIG_SOUND_OSS_CORE_PRECLAIM=y
+	CONFIG_HOSTAUDIO 
   FILES:=$(SOUNDCORE_FILES)
   AUTOLOAD:=$(call AutoLoad,30,$(SOUNDCORE_LOAD))
   $(call AddDepends/input)
Index: package/kernel/modules/netfilter.mk
===================================================================
--- package/kernel/modules/netfilter.mk	(revision 28854)
+++ package/kernel/modules/netfilter.mk	(working copy)
@@ -17,8 +17,8 @@
   	CONFIG_NETFILTER=y \
 	CONFIG_NETFILTER_ADVANCED=y \
 	$(KCONFIG_IPT_CORE)
-  FILES:=$(foreach mod,$(IPT_CORE-m),$(LINUX_DIR)/net/$(mod).ko)
-  AUTOLOAD:=$(call AutoLoad,40,$(notdir $(IPT_CORE-m)))
+  FILES:=$(LINUX_DIR)/net/netfilter/x_tables.ko $(LINUX_DIR)/net/ipv4/netfilter/ip_tables.ko $(LINUX_DIR)/net/ipv4/netfilter/iptable_filter.ko
+  AUTOLOAD:=$(call AutoLoad,40,$(notdir x_tables ip_tables iptable_filter))
 endef
 
 define KernelPackage/ipt-core/description
@@ -45,8 +45,8 @@
 define KernelPackage/ipt-conntrack
   TITLE:=Basic connection tracking modules
   KCONFIG:=$(KCONFIG_IPT_CONNTRACK)
-  FILES:=$(foreach mod,$(IPT_CONNTRACK-m),$(LINUX_DIR)/net/$(mod).ko)
-  AUTOLOAD:=$(call AutoLoad,41,$(notdir $(IPT_CONNTRACK-m)))
+  FILES:=$(LINUX_DIR)/net/netfilter/nf_conntrack.ko $(LINUX_DIR)/net/ipv4/netfilter/nf_defrag_ipv4.ko $(LINUX_DIR)/net/ipv4/netfilter/nf_conntrack_ipv4.ko $(LINUX_DIR)/net/netfilter/xt_state.ko 
+  AUTOLOAD:=$(call AutoLoad,41,$(notdir nf_conntrack nf_defrag_ipv4 nf_conntrack_ipv4 xt_state))
   $(call AddDepends/ipt)
 endef
 
