Index: target/linux/ar71xx/config-2.6.39
===================================================================
--- target/linux/ar71xx/config-2.6.39	(revision 28854)
+++ target/linux/ar71xx/config-2.6.39	(working copy)
@@ -205,3 +205,22 @@
 CONFIG_USB_SUPPORT=y
 CONFIG_XZ_DEC=y
 CONFIG_ZONE_DMA_FLAG=0
+CONFIG_NO_HZ=y
+CONFIG_PREEMPT=y
+# CONFIG_PREEMPT_NONE is not set
+CONFIG_PREEMPT_RCU=y
+# CONFIG_TINY_RCU is not set
+CONFIG_TINY_PREEMPT_RCU=y
+CONFIG_RCU_BOOST=y
+CONFIG_RCU_BOOST_PRIO=1
+CONFIG_RCU_BOOST_DELAY=500
+# CONFIG_INLINE_SPIN_UNLOCK is not set
+# CONFIG_INLINE_SPIN_UNLOCK_IRQ is not set
+# CONFIG_INLINE_READ_UNLOCK is not set
+# CONFIG_INLINE_READ_UNLOCK_IRQ is not set
+# CONFIG_INLINE_WRITE_UNLOCK is not set
+# CONFIG_INLINE_WRITE_UNLOCK_IRQ is not set
+CONFIG_INET_LRO=y
+# CONFIG_NET_SCHED is not set
+# CONFIG_EXT4_FS_XATTR is not set
+# CONFIG_EXT3_FS_XATTR is not set
