Index: feeds/packages/sound/pulseaudio/files/system.pa
===================================================================
--- feeds/packages/sound/pulseaudio/files/system.pa	(revision 0)
+++ feeds/packages/sound/pulseaudio/files/system.pa	(revision 0)
@@ -0,0 +1,62 @@
+#!/usr/bin/pulseaudio -nF
+#
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, write to the Free Software Foundation,
+# Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+
+
+### Load audio drivers statically
+### (it's probably better to not load these drivers manually, but instead
+### use module-udev-detect -- see below -- for doing this automatically)
+#load-module module-alsa-sink sink_name=alsa tsched=0
+#load-module module-alsa-source device=hw:0,0
+#load-module module-null-sink
+#load-module module-pipe-sink
+
+
+
+
+
+### Network access (may be configured with paprefs, so leave this commented
+### here if you plan to use paprefs)
+#load-module module-esound-protocol-tcp
+load-module module-native-protocol-tcp auth-anonymous=1
+load-module module-native-protocol-unix auth-anonymous=1
+
+
+### Automatically move streams to the default sink if the sink they are
+### connected to dies, similar for sources
+load-module module-rescue-streams
+
+### Make sure we always have a sink around, even if it is a null sink.
+load-module module-always-sink
+
+### Automatically suspend sinks/sources that become idle for too long
+load-module module-suspend-on-idle
+
+### Automatically restore the volume of streams and devices
+load-module module-stream-restore
+load-module module-device-restore
+
+### Automatically restore the default sink/source when changed by the user
+### during runtime
+### NOTE: This should be loaded as early as possible so that subsequent modules
+### that look up the default sink/source get the right value
+load-module module-default-device-restore
+
+
+### Make some devices default
+set-default-sink card0
+#set-default-source input
Index: feeds/packages/sound/pulseaudio/files/daemon.conf
===================================================================
--- feeds/packages/sound/pulseaudio/files/daemon.conf	(revision 0)
+++ feeds/packages/sound/pulseaudio/files/daemon.conf	(revision 0)
@@ -0,0 +1,86 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+# USA.
+
+## Configuration file for the PulseAudio daemon. See pulse-daemon.conf(5) for
+## more information. Default values are commented out.  Use either ; or # for
+## commenting.
+
+; daemonize = no
+; fail = yes
+; allow-module-loading = yes
+; allow-exit = yes
+; use-pid-file = yes
+; system-instance = no
+; enable-shm = yes
+; shm-size-bytes = 0 # setting this 0 will use the system-default, usually 64 MiB
+; lock-memory = no
+; cpu-limit = no
+
+; high-priority = yes
+; nice-level = -11
+
+realtime-scheduling = yes
+realtime-priority = 9
+
+; exit-idle-time = 20
+; scache-idle-time = 20
+
+; dl-search-path = (depends on architecture)
+
+; load-default-script-file = yes
+; default-script-file = /etc/pulse/default.pa
+
+; log-target = auto
+; log-level = notice
+; log-meta = no
+; log-time = no
+; log-backtrace = 0
+
+resample-method = trivial
+; resample-method = speex-float-3
+enable-remixing = yes
+; enable-lfe-remixing = no
+
+; flat-volumes = yes
+
+; rlimit-fsize = -1
+; rlimit-data = -1
+; rlimit-stack = -1
+; rlimit-core = -1
+; rlimit-as = -1
+; rlimit-rss = -1
+; rlimit-nproc = -1
+; rlimit-nofile = 256
+; rlimit-memlock = -1
+; rlimit-locks = -1
+; rlimit-sigpending = -1
+; rlimit-msgqueue = -1
+; rlimit-nice = 31
+; rlimit-rtprio = 9
+; rlimit-rttime = 1000000
+
+; default-sample-format = s16le
+; default-sample-rate = 44100
+; default-sample-channels = 2
+; default-channel-map = front-left,front-right
+
+default-fragments = 8
+default-fragment-size-msec = 16
+
+; enable-deferred-volume = yes
+; deferred-volume-safety-margin-usec = 8000
+; deferred-volume-extra-delay-usec = 0
Index: feeds/packages/sound/pulseaudio/files/pulseaudio.init
===================================================================
--- feeds/packages/sound/pulseaudio/files/pulseaudio.init	(revision 28602)
+++ feeds/packages/sound/pulseaudio/files/pulseaudio.init	(working copy)
@@ -3,8 +3,22 @@
 START=65
 
 start() {
+	export HOME=/tmp
 	mkdir -p /tmp/lib/pulse
-	/usr/bin/pulseaudio --daemonize --system --disallow-exit --disallow-module-loading --disable-shm --exit-idle-time=-1
+	/usr/bin/pulseaudio --daemonize --system --disallow-exit --realtime --disable-shm --exit-idle-time=-1 -v
+	for card in /proc/asound/card*; do
+	    device_name=$(echo $card | cut -f4 -d"/")
+	    if [ -e "$card" ] && [ "$device_name" != "cards" ]; then
+	      /usr/bin/env -i ACTION=add DEVICENAME="$device_name" /sbin/hotplug-call sound
+	    fi
+	done
+	for input in /dev/input/event*; do
+	    input_id=$(echo $input | cut -f4 -d"/" | sed -e "s/event//")
+	    VOL_RESULT=$(input-kbd $input_id 2> /dev/null | grep KEY_VOL)
+	    if [ -e "$input" ] && [ "$VOL_RESULT" != "" ]; then
+	      /usr/bin/env -i ACTION=add DEVICENAME="event$input_id" /sbin/hotplug-call input
+	    fi
+	done
 }
 
 stop() {
Index: feeds/packages/sound/pulseaudio/patches/002-remove-ev-bit-check-module-mmkbd-evdev.patch
===================================================================
--- feeds/packages/sound/pulseaudio/patches/002-remove-ev-bit-check-module-mmkbd-evdev.patch	(revision 0)
+++ feeds/packages/sound/pulseaudio/patches/002-remove-ev-bit-check-module-mmkbd-evdev.patch	(revision 0)
@@ -0,0 +1,33 @@
+--- a/src/modules/module-mmkbd-evdev.c	2011-06-24 00:29:11.000000000 +0100
++++ b/src/modules/module-mmkbd-evdev.c.modified	2011-12-11 18:35:20.000000000 +0000
+@@ -195,6 +195,7 @@
+         pa_log("Failed to open evdev device: %s", pa_cstrerror(errno));
+         goto fail;
+     }
++    pa_log("device name: %s", pa_modargs_get_value(ma, "device", DEFAULT_DEVICE));
+ 
+     if (ioctl(u->fd, EVIOCGVERSION, &version) < 0) {
+         pa_log("EVIOCGVERSION failed: %s", pa_cstrerror(errno));
+@@ -217,18 +218,16 @@
+         goto fail;
+     }
+ 
+-    pa_log_info("evdev device name: %s", name);
+-
+     memset(evtype_bitmask, 0, sizeof(evtype_bitmask));
+     if (ioctl(u->fd, EVIOCGBIT(0, EV_MAX), evtype_bitmask) < 0) {
+         pa_log("EVIOCGBIT failed: %s", pa_cstrerror(errno));
+         goto fail;
+     }
+ 
+-    if (!test_bit(EV_KEY, evtype_bitmask)) {
+-        pa_log("Device has no keys.");
+-        goto fail;
+-    }
++    //if (!test_bit(EV_KEY, evtype_bitmask)) {
++    //    pa_log("Device has no keys.");
++    //    goto fail;
++    //}
+ 
+     u->io = m->core->mainloop->io_new(m->core->mainloop, u->fd, PA_IO_EVENT_INPUT|PA_IO_EVENT_HANGUP, io_callback, u);
+ 
Index: feeds/packages/sound/pulseaudio/Makefile
===================================================================
--- feeds/packages/sound/pulseaudio/Makefile	(revision 28602)
+++ feeds/packages/sound/pulseaudio/Makefile	(working copy)
@@ -114,7 +114,13 @@
 	--disable-asyncns \
 	--disable-lirc \
 	--disable-bluez \
-	--without-caps
+	--without-caps \
+	--disable-waveout \
+	--disable-gtk2 \
+	--disable-hal-compat \
+	--disable-ipv6 \
+	--disable-openssl \
+	--without-fftw
 
 ifeq ($(BUILD_VARIANT),avahi)
 CONFIGURE_ARGS += \
@@ -168,18 +174,34 @@
 	$(INSTALL_CONF) \
 		$(PKG_INSTALL_DIR)/etc/pulse/* \
 		$(1)/etc/pulse
+	
+	$(INSTALL_CONF) ./files/system.pa $(1)/etc/pulse
+	$(INSTALL_CONF) ./files/daemon.conf $(1)/etc/pulse
 
-	$(CP) \
-		$(PKG_INSTALL_DIR)/usr/lib/*.so* \
-		$(1)/usr/lib/
+	$(CP) 	$(PKG_INSTALL_DIR)/usr/lib/libpulsecore-1.1.so* $(1)/usr/lib/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/libpulsecommon-1.1.so* $(1)/usr/lib/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/libpulse.so* $(1)/usr/lib/
 
-	$(CP) \
-		$(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/lib*.so \
-		$(1)/usr/lib/
 
-	$(CP) \
-		$(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module*.so \
-		$(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/libprotocol-native.so $(1)/usr/lib/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/libalsa-util.so $(1)/usr/lib/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/librtp.so $(1)/usr/lib/
+
+	$(CP) 	$(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-alsa-sink.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-native-protocol-tcp.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-native-protocol-unix.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-suspend-on-idle.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-rtp-recv.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-rtp-send.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-stream-restore.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-device-restore.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-default-device-restore.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-rescue-streams.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-always-sink.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-null-sink.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-mmkbd-evdev.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-switch-on-connect.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pulse-$(PKG_VERSION)/modules/module-remap-sink.so $(1)/usr/lib/pulse-$(PKG_VERSION)/modules/
 endef
 
 define Package/pulseaudio-daemon-avahi/install
@@ -203,6 +225,9 @@
 		$(PKG_INSTALL_DIR)/etc/pulse/* \
 		$(1)/etc/pulse
 
+	$(INSTALL_CONF) ./files/system.pa $(1)/etc/pulse
+	$(INSTALL_CONF) ./files/daemon.conf $(1)/etc/pulse
+
 	$(CP) \
 		$(PKG_INSTALL_DIR)/usr/lib/*.so* \
 		$(1)/usr/lib/
@@ -229,8 +254,9 @@
 		$(1)/usr/bin
 
 	$(INSTALL_BIN) \
-		$(PKG_INSTALL_DIR)/usr/bin/pa* \
+		$(PKG_INSTALL_DIR)/usr/bin/pactl \
 		$(1)/usr/bin/
+
 endef
 
 define Package/pulseaudio-profiles/install
